### PR TITLE
Updates to use new RuboCop flag.

### LINKED ISSUE #38: Rubocop 0.83 breaking change: only consider known file types
With rubocop version 0.83 they introduced a breaking change that also affects `danger-rubocop`:

https://github.com/rubocop-hq/rubocop/pull/7930

You now have to pass `--only-recognized-file-types`, otherwise danger will fail if one non-ruby-file was changed. If one non-ruby file is found Rubocop still tries to parse it and will fail with tons of errors:

```
app/views/....haml#L1: unexpected token error
spec/..._ctrl_spec.coffee#L2: unexpected token tRPAREN
spec/..._ctrl_spec.coffee#L3: unexpected token tLABEL
```

As a result, we need to pass the mentioned flag to rubocop now.

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #38.

### CODE REVIEW

File: lib/danger_plugin.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def lint(config = nil)
     private
 
     def rubocop(files_to_lint, force_exclusion, only_report_new_offenses, config_path: nil)
-      base_command = ['rubocop', '-f', 'json']
+      base_command = ['rubocop', '-f', 'json', '--only-recognized-file-types']
```
Comment (ashfurrow) [inline lib/danger_plugin.rb:56]: This is the only meaningful change.

**Comment by gbero** [comment]
Many thanks 👍

**Comment by kitop** [comment]
I just noticed this breaks danger-rubocop with any version of rubocop below 0.83, since it breaks with `invalid option: --only-recognized-file-types` error.

Would it be possible to either check for version or add a min version requirement in the gemspec? I'm happy to open a PR for either, but would love to know your preferred option.

**Comment by ashfurrow** [comment]
Oh yikes, hmm. Adding a check would introduce a kind of technical debt for older versions, whose use should only really ever fall. I think a minimum version in the gemspec would be better. Maybe we can do the following:

- Revert this change and release a new patch version of `danger-rubocop`
- Re-apply the change _and_ add the minimum version to the gemspec 
- Release a minor version of `danger-rubocop`

This should help others avoid this mistake, hopefully leap-frogging over the version that introduced this incompatibility. What do you think, @kitop?

**Comment by kitop** [comment]
I think that's a great plan. Just to confirm I understand, this would look like
- Version `0.8.1` reverts this
- Version `0.9.0` introduces this again with the minimum version added in gemspec

That'd be great and provide the best rollforward compatibility IMO.

Thank you so much for the quick response!

**Comment by ashfurrow** [comment]
Yup, exactly. Do you want to open a pull request? I can then handle the first release, the merge, and then subsequent release all at once.

**Comment by kitop** [comment]
Will do! Thanks!

**Comment by ashfurrow** [comment]
Okay! I've got those released to RubyGems and we should be all set. Let me know if it works on your end?

**Comment by kitop** [comment]
0.8.1 worked like a charm!! Thank you so much!

**Comment by ashfurrow** [comment]
Awesome! Thanks for bringing up this issue, for sending the PR, and for following up!

**Pull request merged**